### PR TITLE
Passenger preferences entity

### PR DESCRIPTION
**Pull request by Katyh-one**
Updated the passenger preferences entity to include the relationship with the person id in person. 

Added in extra mappings in service/ controller. 

Updated test data.

### CODE REVIEW

File: src/main/java/com/qa/sky_ride_backend/controllers/PassengerPreferencesController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    public PassengerPreferencesController(PassengerPreferencesService service, PassengerPreferencesRepo repo){
+        super();
+        this.service = service;
+        this.repo = repo;
+    }
+
+    @GetMapping("/getAllPreferences")
+    public List<PassengerPreferencesDTO> getAllPreferences() {
+        List<PassengerPreferences> preferencesList = repo.findAll();
```
Comment (hyperleash) [inline src/main/java/com/qa/sky_ride_backend/controllers/PassengerPreferencesController.java:26]: Controller shouldn't be really talking to the repo directly I think? I just noticed that this is also the same in driverDetails. If it still functional, I think that's okay in the interest of time, but if we will have time to fix it - we can.

File: src/main/java/com/qa/sky_ride_backend/services/PassengerPreferencesService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        this.repo = repo;
+        this.personRepo = personRepo;
+        this.mapper = mapper;
+    }
+
+    private PassengerPreferencesDTO mapToPassengerPreferencesDTO(PassengerPreferences preferences){
+        return this.mapper.map(preferences, PassengerPreferencesDTO.class);
+    }
+
+    public List<PassengerPreferences> getAllPreferences() {
```
Comment (hyperleash) [inline src/main/java/com/qa/sky_ride_backend/services/PassengerPreferencesService.java:30]: This is never used by anything, instead an equivalent method in the Controller used, talking directly to the repo, which is not best practice. So we need to either remove this, or make a call to the controller to this instead.

File: src/main/java/com/qa/sky_ride_backend/services/PassengerPreferencesService.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    private ModelMapper mapper;
+
+    public PassengerPreferencesService(PassengerPreferencesRepo repo, PersonRepo personRepo, ModelMapper mapper) {
+        super();
+        this.repo = repo;
+        this.personRepo = personRepo;
+        this.mapper = mapper;
+    }
+
+    private PassengerPreferencesDTO mapToPassengerPreferencesDTO(PassengerPreferences preferences){
```
Comment (hyperleash) [inline src/main/java/com/qa/sky_ride_backend/services/PassengerPreferencesService.java:26]: same here, this is never used

File: src/main/java/com/qa/sky_ride_backend/controllers/PassengerPreferencesController.java (hunk 1 above)
Comment (Katyh-one) [reply src/main/java/com/qa/sky_ride_backend/controllers/PassengerPreferencesController.java]: Think this should now be resolved on my latest push :)

**Review by hyperleash — APPROVED** [review]

**Pull request merged**